File: 150-EvaluateReversePolishNotation/150-EvaluateReversePolishNotation.java
// Last updated: 04/04/2026, 23:36:30
class Solution {
    public int evalRPN(String[] tokens) {
        int n=tokens.length;
        int[] st=new int[n];
        int top=-1;
        for (String token : tokens) {
            if (token.length()==1) {
                char c=token.charAt(0);
                if (c=='+' || c=='-' || c=='*' || c=='/') {
                    int a=st[top--];
                    int b=st[top--];
                    switch (c) {
                        case '+': st[++top]=b+a; break;
                        case '-': st[++top]=b-a; break;
                        case '*': st[++top]=b*a; break;
                        case '/': st[++top]=b/a; break;
                    }
                    continue;
                }
            }
            st[++top]=Integer.parseInt(token);
        }
        return st[top];
    }
}

File: 150-EvaluateReversePolishNotation/150-EvaluateReversePolishNotation.cpp
// Last updated: 04/04/2026, 23:36:35
class Solution {
public:
    int evalRPN(vector<string>& tokens) {
        vector<int> st;
        st.reserve(tokens.size());
        for (const string& token : tokens) {
            if (token.size()==1 && (token[0]=='+' || token[0]=='-' || 
                                      token[0]=='*' || token[0]=='/')) {
                int a=st.back(); st.pop_back();
                int b=st.back(); st.pop_back();
                switch (token[0]) {
                    case '+': st.push_back(b+a); break;
                    case '-': st.push_back(b-a); break;
                    case '*': st.push_back(b*a); break;
                    case '/': st.push_back(b/a); break;
                }
            } else {
                st.push_back(stoi(token));
            }
        }
        return st.back();
    }
};
